### PR TITLE
Add dask.dataframe.series.view()

### LINKED ISSUE #8439: Series / DataFrame view method not implemented.
When attempting to convert a column from one dtype to another using `astype`, there is a `FutureWarning` raised by Dask and Pandas regarding the move to using `view`.

Pandas suggests replacing the `astype` method with `view`, however `view` appears not to be implemented in `dask.dataframe.core`.

This is the `FutureWarning` from Pandas:

> FutureWarning: casting datetime64[ns] values to int64 with .astype(...) is deprecated and will raise in a future version. Use .view(...) instead.
df['TIMESTAMP_astype'] = df['TIMESTAMP'].astype('int64')

Setup:
```
import pandas as pd
import dask.dataframe as dd

data = {
    'TIMESTAMP': [
        '2021-11-27 00:05:02.175274',
        '2021-11-27 00:05:05.205596',
        '2021-11-27 00:05:29.212572',
        '2021-11-27 00:05:25.708343',
        '2021-11-27 00:05:47.714958',
    ]
}

df = pd.DataFrame(data)
df['TIMESTAMP'] = pd.to_datetime(df['TIMESTAMP'])

ddf = dd.from_pandas(df, npartitions=1)
ddf['TIMESTAMP'] = dd.to_datetime(ddf['TIMESTAMP'])
```

Pandas code:
```
df['TIMESTAMP_astype'] = df['TIMESTAMP'].astype('int64')  # works - throws FutureWarning
df['TIMESTAMP_view'] = df['TIMESTAMP'].view('int64')  # works

assert (df['TIMESTAMP_astype'] == df['TIMESTAMP_view']).all()  # True
```

Dask code:
```
ddf['TIMESTAMP_astype'] = ddf['TIMESTAMP'].astype('int64')  # works - throws FutureWarning
ddf['TIMESTAMP_view'] = ddf['TIMESTAMP'].view('int64')  # error - below
```

Not implemented `view` in `dask.dataframe.core`:
> AttributeError: 'Series' object has no attribute 'view'

### PR DESCRIPTION
**Pull request by pavithraes**
- [x] Closes #8439
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: dask/dataframe/tests/test_dataframe.py [hunk 1]
```diff
@@ -4765,6 +4765,12 @@ def test_nunique(dropna, axis):
     )
 
 
+def test_view():
+    s = pd.Series(range(10), dtype="int8")
+    ds = dd.from_pandas(s, npartitions=2)
+    assert_eq(ds.view("uint8"), s.view("uint8"))
```
Comment (ian-r-rose) [inline dask/dataframe/tests/test_dataframe.py:4771]: It think it's also worth adding a test for the original issue in #8439 (that is to say, that we can `view` a datetime without issue)

File: dask/dataframe/core.py [hunk 2]
```diff
@@ -3839,6 +3839,10 @@ def is_monotonic_decreasing(self):
             token="monotonic_decreasing",
         )
 
+    @derived_from(pd.Series)
+    def view(self, dtype):
+        return self.map_partitions(M.view, dtype, enforce_metadata=False)
```
Comment (ian-r-rose) [inline dask/dataframe/core.py:3844]: You may need to also update the `_meta` object for consistency

**Review by ian-r-rose — COMMENTED** [review]
Thanks @pavithraes, this is looking good!

**Review by ian-r-rose — APPROVED** [review]
Thanks @pavithraes, this looks good to me!

**Comment by jsignell** [comment]
add to allowlist

**Comment by jsignell** [comment]
This is in! Thanks @pavithraes :)

**Pull request merged**